spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/EventHubTemplate.java

```diff
@@ -124,18 +124,19 @@ public boolean subscribe(String destination, String consumerGroup, Consumer<Mess
     public boolean unsubscribe(String destination, String consumerGroup) {
         Tuple<String, String> nameAndConsumerGroup = Tuple.of(destination, consumerGroup);
 
-        if (!processorHostsByNameAndConsumerGroup.containsKey(nameAndConsumerGroup)) {
+        if (!processorByNameAndGroup.containsKey(nameAndConsumerGroup)) {
             return false;
         }
 
-        processorHostsByNameAndConsumerGroup.remove(nameAndConsumerGroup).unregisterEventProcessor()
-                                            .whenComplete((s, t) -> {
-                                                if (t != null) {
-                                                    LOGGER.warn(String.format(
-                                                            "Failed to unregister consumer '%s' with group '%s'",
-                                                            destination, consumerGroup), t);
-                                                }
-                                            });
+        EventProcessorHost processorHost = processorByNameAndGroup.remove(nameAndConsumerGroup);
+
+        processorHost.unregisterEventProcessor().whenComplete((s, t) -> {
```

Comment:
the thread will return without waiting for the action to complete?

Reply:
Yes